runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java

```diff
@@ -401,8 +402,34 @@ static boolean isServiceEndpoint(String endpoint) {
     return Strings.isNullOrEmpty(endpoint) || Pattern.matches(ENDPOINT_REGEXP, endpoint);
   }
 
+  static void validateSdkContainerImageOptions(DataflowPipelineWorkerPoolOptions workerOptions) {
+    // Check against null - empty string value for workerHarnessContainerImage
+    // must be preserved for legacy dataflowWorkerJar to work.
+    Preconditions.checkArgument(
+        workerOptions.getWorkerHarnessContainerImage() == null
+            || workerOptions.getSdkContainerImage() == null
+            || workerOptions
+                .getWorkerHarnessContainerImage()
+                .equals(workerOptions.getSdkContainerImage()),
+        "Cannot use legacy option workerHarnessContainerImage with sdkContainerImage. Prefer sdkContainerImage.");
+
+    // Make sure the values are the same for legacy flag and current flag.
+    if (workerOptions.getSdkContainerImage() != null) {
+      workerOptions.setWorkerHarnessContainerImage(workerOptions.getSdkContainerImage());
+    } else if (workerOptions.getWorkerHarnessContainerImage() != null) {
+      LOG.warn(
+          "Prefer --sdkContainerImage over deprecated legacy option --workerHarnessContainerImage.");
+      workerOptions.setSdkContainerImage(workerOptions.getWorkerHarnessContainerImage());
+    }
+  }
```

Comment:
Is it possible that both sdkContainerImage and workerHarnessContainerImage are null?

Reply:
yup, handled by getContainerImageForJob - if sdkContainerImage is null, it uses the default image